src/python/pants/build_graph/address.py

```diff
@@ -257,28 +268,38 @@ class BuildFileAddress(Address):
   :API: public
   """
 
-  def __init__(self, build_file=None, target_name=None, rel_path=None):
+  def __init__(
+    self,
+    build_file: Optional[BuildFile] = None,
+    target_name: Optional[str] = None,
+    rel_path: Optional[str] = None
+  ) -> None:
     """
     :param build_file: The build file that contains the object this address points to.
-    :type build_file: :class:`pants.base.build_file.BuildFile`
-    :param string rel_path: The BUILD files' path, relative to the root_dir.
-    :param string target_name: The name of the target within the BUILD file; defaults to the default
-                               target, aka the name of the BUILD file parent dir.
+    :param rel_path: The BUILD files' path, relative to the root_dir.
+    :param target_name: The name of the target within the BUILD file; defaults to the default
+                        target, aka the name of the BUILD file parent dir.
 
     :API: public
     """
-    rel_path = rel_path or build_file.relpath
```

Comment:
This had an edge case that could fail if _both_ `rel_path` and `build_file` were None (the defaults). We now check for this.